store/postgres/tests/relational_bytes.rs

```diff
@@ -0,0 +1,303 @@
+    let mut scrubbed = Entity::new();
+    // merge has the sideffect of removing any attribute
+    // that is Value::Null
+    scrubbed.merge(entity.clone());
+    scrubbed
+}
+
+macro_rules! assert_entity_eq {
+    ($left:expr, $right:expr) => {{
+        let (left, right) = (&($left), &($right));
+        let mut pass = true;
+
+        for (key, left_value) in left.iter() {
+            match right.get(key) {
+                None => {
+                    pass = false;
+                    println!("key '{}' missing from right", key);
+                }
+                Some(right_value) => {
+                    if left_value != right_value {
+                        pass = false;
+                        println!(
+                            "values for '{}' differ:\n     left: {:?}\n    right: {:?}",
+                            key, left_value, right_value
+                        );
+                    }
+                }
+            }
+        }
+        for key in right.keys() {
+            if left.get(key).is_none() {
+                pass = false;
+                println!("key '{}' missing from left", key);
+            }
+        }
+        assert!(pass, "left and right entities are different");
+    }};
+}
+
+/// Test harness for running database integration tests.
+fn run_test<R, F>(test: F)
+where
+    F: FnOnce(&PgConnection, &Layout) -> R + Send + 'static,
+    R: IntoFuture<Item = ()> + Send + 'static,
+    R::Error: Send + Debug,
+    R::Future: Send,
+{
+    let url = postgres_test_url();
+    let conn = PgConnection::establish(url.as_str()).expect("Failed to connect to Postgres");
+
+    // Lock regardless of poisoning. This also forces sequential test execution.
+    let mut runtime = match STORE_RUNTIME.lock() {
+        Ok(guard) => guard,
+        Err(err) => err.into_inner(),
+    };
+
+    runtime
+        .block_on(async {
+            // Reset state before starting
+            remove_test_data(&conn);
+
+            // Seed database with test data
+            let layout = create_schema(&conn);
+
+            // Run test
+            test(&conn, &layout).into_future().compat().await
+        })
+        .expect("Failed to run ChainHead test");
+}
+
+#[test]
+fn bad_id() {
+    run_test(|conn, layout| -> Result<(), ()> {
+        // We test that we get errors for various strings that are not
+        // valid 'Bytes' strings; we use `find` to force the conversion
+        // from String -> Bytes internally
+        let res = layout.find(conn, "Thing", "bad", BLOCK_NUMBER_MAX);
+        assert!(res.is_err());
+        assert_eq!(
+            "store error: Odd number of digits",
+            res.err().unwrap().to_string()
+        );
+
+        // We do not allow the `\x` prefix that Postgres uses
+        let res = layout.find(conn, "Thing", "\\xbadd", BLOCK_NUMBER_MAX);
+        assert!(res.is_err());
+        assert_eq!(
+            "store error: Invalid character \'\\\\\' at position 0",
```

Comment:
Who's escaping whom? 🤣

Reply:
Haha .. yes, counting backslashes is my new hobby

Reply:
🤣